src/pages/evm/nfts/NftInventoryPage.vue

```diff
@@ -101,16 +109,70 @@ const tableRows = computed(() => {
         collectionAddress: nft.contractAddress,
     }));
 });
-const listImagesLoadingStates = ref<Record<string, boolean>>({});
+const showNoFilteredResultsState = computed(() => (collectionFilter.value || searchFilter.value) && !nftsToShow.value.length);
 
 
 // watchers
+watch(nftsAndCollectionListLoaded, (loaded) => {
+    if (loaded) {
+        nfts.value.forEach((nft) => {
+            // if an NFT icon in list view hasn't yet been loaded, enable the loading state for that image
+            // the loading state will be ended when the @loaded event is fired by that image
+            if (!showNftsAsTiles.value && listImagesLoadingStates.value?.[nft.id] !== false) {
+                listImagesLoadingStates.value[nft.id] = true;
+            }
+        });
+
+        // if this is initial load and there are pagination query params, validate and apply them
+        const { rowsPerPage, page, collection, search } = route.query;
+
+        let newRowsPerPage = 12;
+        let newPage = 1;
+
+        if (rowsPerPage && rowsPerPageOptions.includes(+(rowsPerPage ?? 12))) {
+            newRowsPerPage = +(rowsPerPage ?? 12);
+        }
+
+        // if the page query param is valid (not greater than the total number of pages), apply it
+        if (page && (newRowsPerPage * (+(page ?? 1) - 1)) < nfts.value.length) {
+            newPage = +(page ?? 1);
```

Comment:
is this conditional assignment necessary? add comments to clarify if condition for readability

Reply:
will do, thanks